addons/mail/static/src/new/command_palette.js

```diff
@@ -0,0 +1,83 @@
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+import { _lt } from "@web/core/l10n/translation";
+import { cleanTerm } from "@mail/new/utils/format";
+import { Component, xml } from "@odoo/owl";
+
+const commandSetupRegistry = registry.category("command_setup");
+const commandProviderRegistry = registry.category("command_provider");
+
+// -----------------------------------------------------------------------------
+// add @ namespace + provider
+// -----------------------------------------------------------------------------
+commandSetupRegistry.add("@", {
+    debounceDelay: 200,
+    emptyMessage: _lt("No user found"),
+    name: _lt("users"),
+    placeholder: _lt("Search for a user..."),
+});
+
+class DialogCommand extends Component {}
+DialogCommand.template = xml`
+    <div class="o_command_default d-flex align-items-center justify-content-between px-4 py-2 cursor-pointer">
+        <t t-slot="name"/>
+        <span t-if="props.email" t-out="props.email"/>
+    </div>
+`;
+
+commandProviderRegistry.add("mail.partner", {
+    namespace: "@",
+    async provide(env, options) {
+        /** @type {import("@mail/new/core/messaging").Messaging} */
+        const messaging = env.services["mail.messaging"];
+        return messaging.searchPartners(options.searchValue).then((results) => {
```

Comment:
Any particular reason to use `then` rather than `await` (while keeping `async` on the function as well)?